x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/response/cohere/CohereRankedResponseEntity.java

```diff
@@ -142,7 +146,18 @@ private static RankedDocsResults.RankedDoc parseRankedDocObject(XContentParser p
                 parser.nextToken();
             }
         }
-        return new RankedDocsResults.RankedDoc(String.valueOf(index), String.valueOf(relevanceScore), String.valueOf(documentText));
+
+        if (index == -1) {
+            logger.error("Failed to find required field [index] in Cohere embeddings response");
+        }
+        if (relevanceScore == -1) {
+            logger.error("Failed to find required field [relevance_score] in Cohere embeddings response");
+        }
+        if (documentText == null) {
+            logger.error("Failed to find required field [document] in Cohere embeddings response");
+        }
+
+        return new RankedDocsResults.RankedDoc(index, relevanceScore, String.valueOf(documentText));
```

Comment:
Do we need the `String.valueOf(documentText)`? It's already a string right?

Reply:
I think it was a mistake to use valueOf there.

Reply:
> I think it was a mistake to use valueOf there.

👍 I left a similar suggestion